Brokerages/Oanda/OandaBrokerage.cs

```diff
@@ -0,0 +1,1052 @@
+                    return (T)serializer.ReadObject(stream);
+                }
+            }
+            catch (WebException ex)
+            {
+                var stream = GetResponseStream(ex.Response);
+                var reader = new StreamReader(stream);
+                var result = reader.ReadToEnd();
+                throw new Exception(result);
+            }
+        }
+
+        /// <summary>
+        /// Secondary (internal) request handler. differs from primary in that parameters are placed in the body instead of the request string
+        /// </summary>
+        /// <typeparam name="T">response type</typeparam>
+        /// <param name="method">method to use (usually POST or PATCH)</param>
+        /// <param name="requestParams">the parameters to pass in the request body</param>
+        /// <param name="requestString">the request to make</param>
+        /// <returns>response, via type T</returns>
+        private T MakeRequestWithBody<T>(string requestString, string method, Dictionary<string, string> requestParams)
+        {
+            // Create the body
+            var requestBody = CreateParamString(requestParams);
+            var request = WebRequest.CreateHttp(requestString);
+            request.Headers[HttpRequestHeader.Authorization] = "Bearer " + AccessToken;
+            request.Method = method;
+            request.ContentType = "application/x-www-form-urlencoded";
+
+            using (var writer = new StreamWriter(request.GetRequestStream()))
+            {
+                // Write the body
+                writer.WriteAsync(requestBody);
+            }
+
+            // Handle the response
+            try
+            {
+                using (var response = request.GetResponse())
+                {
+                    var serializer = new DataContractJsonSerializer(typeof(T));
+                    return (T)serializer.ReadObject(response.GetResponseStream());
+                }
+            }
+            catch (WebException ex)
+            {
+                var response = (HttpWebResponse)ex.Response;
+                var stream = new StreamReader(response.GetResponseStream());
+                var result = stream.ReadToEnd();
+                throw new Exception(result);
+            }
+        }
+
+        private List<Position> GetPositions()
+        {
+            using (var task = GetPositionsAsync(AccountId))
+            {
+                task.Wait();
+                var result = task.Result;
+                return result;
+            }
+        }
+
+        /// <summary>
+        /// Retrieves the current non-zero positions for a given account
+        /// </summary>
+        /// <param name="accountId">positions will be retrieved for this account id</param>
+        /// <returns>List of Position objects with the details for each position (or empty list iff no positions)</returns>
+        public async Task<List<Position>> GetPositionsAsync(int accountId)
```

Comment:
this method looks like it will always run synchronously, there's no 'await' keywords in the method body. do we intended to call `MakeRequestAsync<T>` instead?